dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/constraints/DatadogDataConstraints.kt

```diff
@@ -32,32 +32,49 @@ internal class DatadogLogConstraints : LogConstraints {
         return convertedTags.take(MAX_TAG_COUNT)
     }
 
-    override fun validateAttributes(attributes: Map<String, Any?>): Map<String, Any?> {
+    override fun validateAttributes(
+        attributes: Map<String, Any?>,
+        maxKeyDepthLevel: Int,
+        attributesGroupName: String?
+    ): Map<String, Any?> {
         val convertedAttributes = attributes.mapNotNull {
-            val key = convertAttributeKey(it.key)
-            if (key == null) {
+            if (it.key == null) {
                 devLogger.e("\"$it\" is an invalid attribute, and was ignored.")
                 null
-            } else {
-                if (key != it.key) {
-                    devLogger.w(
-                        "attribute \"${it.key}\" " +
-                            "was modified to \"$key\" to match our constraints."
-                    )
-                }
-                key to it.value
             }
+            val key = convertAttributeKey(it.key, maxKeyDepthLevel)
+            if (key != it.key) {
+                devLogger.w(
+                    "Key \"${it.key}\" " +
+                        "was modified to \"$key\" to match our constraints."
+                )
+            }
+            key to it.value
         }
         val discardedCount = convertedAttributes.size - MAX_ATTR_COUNT
```

Comment:
i guess we should use `(MAX_ATTR_COUNT + 3)` for `userInfo.extraInfo` because 3 other `usr.***` attributes come from `userInfo` itself

Reply:
Hmmm I see your point in here even though I don't know how this rule applies when we have this case but it doesn't apply for all the event types. For RUM for example the extra attributes are going into the `context.usr` group which is different than the `usr` group where the name, email and id are going. This rule is a bit awkward for me, do you guys have it on iOS too ?. It might make sense to have it for "custom attributes" but not for user info or timings.